### ISSUE TITLE
Resolving of latest version using multiple release branches unexpected

### ISSUE THREAD

**Issue by wolf-michl** [google-github-actions/release-please-action#528]
### TL;DR

When using multiple release branches the resolving of the latest tag is done from the current branch instead of from the whole repository.

### Expected behavior

We are experiencing issues when using multiple release branches.
According to [supporting multiple release branches](https://github.com/google-github-actions/release-please-action#supporting-multiple-release-branches) multiple branches are supported. Currently we are adding `default-branch: ${{ github.ref_name }}` when running on a release branch to ensure that we support all created release branches.
We expected release-please to resolve the version from the whole repository. Otherwise we are not able to use it, as we are currently absolutely locked into using multiple release branches at the same time.

### Observed behavior

The problem is that the latest version is seamingly resolved from the current release branch, which isn't what we need. We may have multiple release branches at the same time, so one may have a newer tag than the one which release-please currently runs for.

### Action YAML

```yaml
name: Release flow test
on:
  workflow_dispatch:
  push:
    branches-ignore:
      - release-please*
  pull_request: {}

jobs:
  release-please:
    runs-on: ubuntu-latest
    name: Release please
    outputs:
      release_created: ${{ steps.release_please.outputs.release_created }}
      tag_name: ${{ steps.release_please.outputs.tag_name }}
    steps:
      - name: Release please
        id: release_please
        if: startsWith(github.ref, 'refs/heads/release-')
        uses: google-github-actions/release-please-action@v3
        with:
          release-type: go
          package-name: prototype-versioning
          default-branch: ${{ github.ref_name }}
```

### Additional information

We tried some workarounds for this:

1. After the release was created we merged the release branch using the ours strategy. This was to add the created tag to the default branch where release branches are branched from. This worked, but had the problem of still needing to use a release-as commit to bump versions in release branches, that were created before the latest tag was added to the default branch.
2. After release-please created a PR we create a release-as commit on our default branch with a new version which should be used by release-please in the future (for the next release branch). This works for the version, but not for the changelog, as the commits will still be resolved the same way. This could probably solved using last-release-sha, which isn't something we can add to the action, only in the manifest...

**Comment by chingor13** [comment]
Sorry for the delay in getting to this issue.

If you are not using manifest configs/version file which explicitly declare the latest version, then release-please needs to [figure out what the latest version on that branch was](https://github.com/googleapis/release-please/blob/7c283cadb3651d47e49ebd3e49d371ae952e254b/src/manifest.ts#L1349-L1355). The first attempt is to look through the latest commit history to that branch and find a release-please PR. If we don't find one, then it looks at releases and tags to find the highest semver tag of commit SHAs we've seen on that branch. Do those alternative release branches have release-please PRs or releases tagged?

Without an explicit example, there's not a whole lot we can do to help debug this.

**Comment by wolf-michl** [comment]
@chingor13 
No worries, thanks for getting to it.
Yes, we do not explicitly declare the latest version, as this would kind of defeat the purpose of using release-please in the first place.
The code you linked shows me that this is actually wanted behaviour: release-please only looks for the latest evrsion in the current branch. This what our problem is, as we have multiple release branches (even possibly in parallel), which don't know about each others version, as we knowingly never merge the branches back to our default branch.
Obviopusly this is wanted, but that defeats the purpose of multiple release branches, specifically the default-branch option supporting wildcards.
Ofc if you want I could provide an explicit example for what we are trying to do, but the provided action YAML already contains the necessary information. If you create multiple branches then they will never create releases with subsequent version numbers, so branch 1 creates a 1.0.0 release, but branch 2 also tries to create a 1.0.0, despite it already existing. Just let me know if you need an explicit example or any other additional informations from my site.

**Comment by chingor13** [comment]
Ah, I see. Our primary use case for multiple release branches is for handling LTS/backport versions. In our usage: `main` contains the latest version (say 2.1.3). We'll also have a `1.x` branch that branched from the last v1 release tag. In that branch, we'll configure the `always-bump-patch` versioning strategy, as we know we're only going to add bugfixes to that branch. If necessary, we can add additional versioning strategies as that's a configurable interface that we expose.

We don't have any implementation for `release-please` knowing about versions in other branches, because it's unclear how we would handle that kind of scenario anyways. In your example, both branches have commits that would merit a 1.0.0 version. How would you handle it if you were doing it manually?

**Comment by wolf-michl** [comment]
I would use `git tag -l v*` to get the list of all tags on a repository and then I would resolve the highest semantic version from this list to be used as the base version to calculate a new tag based on semver.

**Comment by chingor13** [comment]
Please re-open this on the primary https://github.com/googleapis/release-please if this is still a feature request.
